src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/RegistrarManager.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using Orleans.GrainDirectory;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    /// <summary>
+    /// Maps multi-cluster registration strategies to the corresponding registrar
+    /// </summary>
+    internal class RegistrarManager
```

Comment:
Do we even need `RegistrarManager` at this point? All its usages appear to be for local cases. Can't we introduce it in a later PR when it and if it's actually needed? This is related to the above question about sync/async `ClusterLocalRegistrar`.